### PR TITLE
Dkrutenko

### PR DESCRIPTION
**Pull request by dkrutenko**
Copleted and working tests according to the code Convention

### CODE REVIEW

File: src/test/java/com/softserveinc/ita/dkrutenko/rozetkaCheckCartTest.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        driver = new ChromeDriver();
+        //page load time
+        driver.manage().timeouts().pageLoadTimeout(10, TimeUnit.SECONDS);
+        //wait amout of time before it throws "No such Element Exception
+        driver.manage().timeouts().implicitlyWait(10, TimeUnit.SECONDS);
+        //Maximize chrome window
+        driver.manage().window().maximize();
+    }
+    //implicitly wait for 2 seconds and quit the driver(close chrome window)
+    @AfterClass(alwaysRun = true)
```
Comment (Nikodych) [inline src/test/java/com/softserveinc/ita/dkrutenko/rozetkaCheckCartTest.java:36]: after method should be at the bottom of the class

File: src/test/java/com/softserveinc/ita/dkrutenko/rozetkaCheckCartTest.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    public void afterClass() {
+        driver.manage().timeouts().implicitlyWait(2, TimeUnit.SECONDS);
+        driver.quit();
+    }
+    //load google.com and delete cookies before work
```
Comment (Nikodych) [inline src/test/java/com/softserveinc/ita/dkrutenko/rozetkaCheckCartTest.java:41]: same

File: src/test/java/com/softserveinc/ita/dkrutenko/rozetkaCheckCartTest.java [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    @BeforeMethod
+    public void beforeMethod() {
+        //Launching actual url
+        driver.get(URL);
+        driver.manage().deleteAllCookies();
+    }
+    //print some message after test finish
```
Comment (Nikodych) [inline src/test/java/com/softserveinc/ita/dkrutenko/rozetkaCheckCartTest.java:48]: same

File: src/test/java/com/softserveinc/ita/dkrutenko/rozetkaCheckCartTest.java [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    @AfterMethod
+    public void afterMethod(ITestResult result) {
+        if(!result.isSuccess()) {
+            String testName = result.getName();
+            System.out.println("Congratulations, " + testName + " has finished working.");
```
Comment (Nikodych) [inline src/test/java/com/softserveinc/ita/dkrutenko/rozetkaCheckCartTest.java:53]: sysout's not allowed in PRs, this is noted in the convention

File: src/test/java/com/softserveinc/ita/dkrutenko/rozetkaCheckCartTest.java [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+        }
+    }
+    //required items for search
```
Comment (Nikodych) [inline src/test/java/com/softserveinc/ita/dkrutenko/rozetkaCheckCartTest.java:56]: same

File: src/test/java/com/softserveinc/ita/dkrutenko/rozetkaCheckCartTest.java [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+    @DataProvider
+    public Object[][] rozetkaCheckCartItems () {
+        return new Object[][]{
+                {"samsung", "Samsung Galaxy S21"},
+                {"iphone",   "iPhone 12 Pro Max"},
+                {"samsung", "Samsung Galaxy A72"},
+
+        };
+    }
+    //Our test with DataProvider that have required items for search
```
Comment (Nikodych) [inline src/test/java/com/softserveinc/ita/dkrutenko/rozetkaCheckCartTest.java:66]: same

File: src/test/java/com/softserveinc/ita/dkrutenko/rozetkaCheckCartTest.java [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+        }
+    }
+    //required items for search
+    @DataProvider
+    public Object[][] rozetkaCheckCartItems () {
+        return new Object[][]{
+                {"samsung", "Samsung Galaxy S21"},
+                {"iphone",   "iPhone 12 Pro Max"},
+                {"samsung", "Samsung Galaxy A72"},
+
```
Comment (Nikodych) [inline src/test/java/com/softserveinc/ita/dkrutenko/rozetkaCheckCartTest.java:63]: delete empty line

File: src/test/java/com/softserveinc/ita/dkrutenko/rozetkaCheckCartTest.java [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+        Assert.assertTrue(phone.contains(item));
+        //click on context menu button
+        driver.findElement(By.xpath("//button[@class='button button--white button--small context-menu__toggle']")).click();
+        //click on "remove" button
+        driver.manage().timeouts().pageLoadTimeout(6,TimeUnit.SECONDS);
+        driverWait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//button[@class='button button--medium button--with-icon button--link context-menu-actions__button']"))).click();
+        //close current windows
+        driver.manage().timeouts().pageLoadTimeout(6,TimeUnit.SECONDS);
+        driverWait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//button[@class='modal__close ng-star-inserted']"))).click();
+
```
Comment (Nikodych) [inline src/test/java/com/softserveinc/ita/dkrutenko/rozetkaCheckCartTest.java:107]: delete empty line

File: src/test/java/com/softserveinc/ita/dkrutenko/rozetkaCheckCartTest.java (hunk 4 above)
Comment (dkrutenko) [reply src/test/java/com/softserveinc/ita/dkrutenko/rozetkaCheckCartTest.java:53]: Sysouts was for me and to show result after test. I know that it is not in convention... Will fix this comments soon

File: src/main/java/com/softserveinc/ita/dkrutenko/pageobjects/BasePage.java [hunk 9]
```diff
@@ -61,17 +48,20 @@ public void clickUserButton() {
...
         return driverWait.until(visibilityOfElementLocated(locator));
     }
 
     public List<WebElement> waitOnElementsList(By locator, int amount) {
-
         return  driverWait.until(numberOfElementsToBeMoreThan(locator, amount));
     }
+
+    public void fillSearchField(String text) {
+        driver.findElement(searchFieldSelector).click();
```
[the remark this answers is not in the document]
Comment (dkrutenko) [reply src/main/java/com/softserveinc/ita/dkrutenko/pageobjects/BasePage.java:63]: you mean var "some name" = driver.findElement(searchFieldSelector) ?

File: src/test/java/com/softserveinc/ita/dkrutenko/rozetkaTests.java [hunk 10]
```diff
@@ -31,15 +31,15 @@ public void verifySearchTest(String brandNameItem, String requiredItem) {
...
     @Test(dataProvider = "rozetkaProductItems")
     public void verifyShoppingCartTest(String brandNameItem, String requiredItem) {
         //fill search with some brand name, click search button and click on required product
         searchGoods.fillSearchField(brandNameItem);
         searchGoods.clickSearchButton();
-        searchGoods.getActualItem(requiredItem).click();
+        searchGoods.getActualItem(requiredItem);
         //add to cart, click continue and go on main page
         shoppingCartPage.clickAddToCartButton();
         shoppingCartModal.clickCartContinueButton();
```
[the remark this answers is not in the document]
Comment (dkrutenko) [reply src/test/java/com/softserveinc/ita/dkrutenko/rozetkaTests.java:45]: click addtocart and clickcartcontinue??? If you are talking about that methods - no, cart isnt empty. Its just additional functionality. In this methods i add to cart some item and click *continue shopping* after that clicking on rozetka logo and going on cart again. I can add same assertTrue with required item, but it is superfluous

[the remark this answers is not in the document]
Comment (dkrutenko) [reply src/test/java/com/softserveinc/ita/dkrutenko/rozetkaTests.java:45]: when I click the "Add to Cart" button, a basket with the current item are opening and i just click "continue" and reopen this basket from main page. I can remove this functions and use assert after adding it to the basket, momentally

File: src/main/java/com/softserveinc/ita/dkrutenko/pageobjects/BasePage.java [hunk 11]
```diff
@@ -59,9 +59,13 @@ public WebElement waitForElementVisibility(By locator) {
...
 
+    public WebElement setSearch() {
+        return driver.findElement(searchFieldSelector);
+    }
+
     public void fillSearchField(String text) {
-        driver.findElement(searchFieldSelector).click();
-        driver.findElement(searchFieldSelector).clear();
-        driver.findElement(searchFieldSelector).sendKeys(text);
+        setSearch().click();
```
Comment (Nikodych) [inline src/main/java/com/softserveinc/ita/dkrutenko/pageobjects/BasePage.java:67]: and what you have changed with such implementation? you look at the same element three times and init three times

I mean such implementation:

var element = driver.findElement(searchFieldSelector);

element.click();
element.clear();
element.sendKeys(text);

File: src/main/java/com/softserveinc/ita/dkrutenko/pageobjects/LoginPageModal.java [hunk 12]
```diff
@@ -35,15 +36,23 @@ public void clickExitButton() {
...
 
+    public WebElement waitForEmailInputElement() {
+        return waitForClickabelElement(emailInputSelector);
+    }
+
     public void fillEmailField(String text) {
-        waitForClickabelElement(emailInputSelector).click();
-        waitForClickabelElement(emailInputSelector).clear();
-        waitForClickabelElement(emailInputSelector).sendKeys(text);
+        waitForEmailInputElement().click();
```
Comment (Nikodych) [inline src/main/java/com/softserveinc/ita/dkrutenko/pageobjects/LoginPageModal.java:44]: same

File: src/main/java/com/softserveinc/ita/dkrutenko/pageobjects/LoginPageModal.java [hunk 13]
```diff
@@ -35,15 +36,23 @@ public void clickExitButton() {
...
+    public WebElement waitForPasswordInputElement() {
+        return waitForClickabelElement(passwordInputSelector);
     }
 
     public void fillPasswordField(String text) {
-        waitForClickabelElement(passwordInputSelector).click();
-        waitForClickabelElement(passwordInputSelector).clear();
-        waitForClickabelElement(passwordInputSelector).sendKeys(text);
+        waitForPasswordInputElement().click();
+        waitForPasswordInputElement().clear();
```
Comment (Nikodych) [inline src/main/java/com/softserveinc/ita/dkrutenko/pageobjects/LoginPageModal.java:55]: same

File: src/test/java/com/softserveinc/ita/dkrutenko/rozetkaTests.java (hunk 10 above)
[the remark this answers is not in the document]
Comment (Nikodych) [reply src/test/java/com/softserveinc/ita/dkrutenko/rozetkaTests.java:45]: leave as is

**Review by Nikodych — CHANGES_REQUESTED** [review]

File: src/main/java/com/softserveinc/ita/dkrutenko/pageobjects/utils/runners/TestRunner.java [hunk 14]
```diff
@@ -0,0 +1,58 @@
+package com.softserveinc.ita.dkrutenko.pageobjects.utils.runners;
```
Comment (Nikodych) [inline src/main/java/com/softserveinc/ita/dkrutenko/pageobjects/utils/runners/TestRunner.java:1]: move utils package from pageobjects package

**Review by Nikodych — CHANGES_REQUESTED** [review]

File: src/test/java/com/softserveinc/ita/dkrutenko/rozetkaTests.java (hunk 10 above)
[the remark this answers is not in the document]
Comment (dkrutenko) [reply src/test/java/com/softserveinc/ita/dkrutenko/rozetkaTests.java:45]: good

File: src/main/java/com/softserveinc/ita/dkrutenko/pageobjects/utils/runners/TestRunner.java (hunk 14 above)
Comment (dkrutenko) [reply src/main/java/com/softserveinc/ita/dkrutenko/pageobjects/utils/runners/TestRunner.java:1]: done

File: src/main/java/com/softserveinc/ita/dkrutenko/pageobjects/BasePage.java (hunk 11 above)
Comment (dkrutenko) [reply src/main/java/com/softserveinc/ita/dkrutenko/pageobjects/BasePage.java]: done

File: src/main/java/com/softserveinc/ita/dkrutenko/pageobjects/utils/runners/TestRunner.java (hunk 14 above)
Comment (Nikodych) [reply src/main/java/com/softserveinc/ita/dkrutenko/pageobjects/utils/runners/TestRunner.java:1]: it still there

**Review by Nikodych — CHANGES_REQUESTED** [review]

File: src/main/java/com/softserveinc/ita/dkrutenko/pageobjects/utils/runners/TestRunner.java (hunk 14 above)
Comment (dkrutenko) [reply src/main/java/com/softserveinc/ita/dkrutenko/pageobjects/utils/runners/TestRunner.java]: it didnt pushed... Dont know why... Now its fine

**Review by Nikodych — APPROVED** [review]

**Pull request merged**